extensions/amp-story/0.1/related-articles.js

```diff
@@ -54,13 +54,23 @@ function buildArticleFromJson_(articleJson) {
     return null;
   }
 
-  user().assert(isProtocolValid(articleJson['url']),
-      `Unsupported protocol for article URL ${articleJson['url']}`);
+  const articleUrl = dev().assert(articleJson['url']);
+  user().assert(isProtocolValid(articleUrl),
+      `Unsupported protocol for article URL ${articleUrl}`);
+
+  let domain;
+  try {
+    domain = getSourceOrigin(articleUrl);
+    domain = removeProtocol(domain);
```

Comment:
Instead, I think we just want to use `parseUrl` from `url.js` and get the hostname.